web/client/components/map/openlayers/DrawSupport.jsx

```diff
@@ -105,20 +109,26 @@ class DrawSupport extends React.Component {
                 if (f.style) {
                     let olFeature = this.toOlFeature(f);
                     if (olFeature) {
-                        olFeature.setStyle(this.toOlStyle(f.style, f.selected));
+                        olFeature.setStyle(f.style && f.style.type ? VectorStyle.getStyle(f) : this.toOlStyle(f.style, f.selected));
```

Comment:
For the future: we should try to remove toOlStyle if possible and always use VectorStyle.getStyle